sample/authorization-code-flow-sample/authorization_code_flow_sample.py

```diff
@@ -0,0 +1,80 @@
+"""
+The configuration file would look like this:
+
+{
+    "authority": "https://login.microsoftonline.com/organizations",
+    "client_id": "your_client_id",
+    "scope": ["https://graph.microsoft.com/.default"],
+    "redirect_uri": "http://localhost:5000/getAToken",
+        // Configure this redirect uri for this sample
+    "client_secret": "yoursecret"
+}
+
+You can then run this sample with a JSON configuration file:
+
+    python sample.py parameters.json
+"""
+
+import sys  # For simplicity, we'll read config file from 1st CLI param sys.argv[1]
+import json
+import logging
+import uuid
+
+import flask as flask
+
+import msal
+
+app = flask.Flask(__name__)
+app.debug = True
+app.secret_key = 'development'
+
+
+# Optional logging
+# logging.basicConfig(level=logging.DEBUG)
+
+config = json.load(open(sys.argv[1]))
+
+application = msal.ConfidentialClientApplication(
+        config["client_id"], authority=config["authority"],
+        client_credential=config["client_secret"],
+        # token_cache=...  # Default cache is in memory only.
+        # You can learn how to use SerializableTokenCache from
+        # https://msal-python.rtfd.io/en/latest/#msal.SerializableTokenCache
+    )
+
+
+@app.route("/")
+def main():
+    login_url = 'http://localhost:5000/login'
```

Comment:
We can use `/login` here, so that we do not need to hard-code the port number here. Otherwise app dev won't immediately know they need to change this line, should they choose to run/deploy this sample on a different port.

BTW, since we are going to change this line anyway, we can probably just do one inline assignment at line 50 below. But this suggestion is totally optional. :-)

Reply:
should this be defined in the config file? or wouldn't the web service itself have this defined somewhere. (is that what /login means?)

---
In reply to: [277909842](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/40#discussion_r277909842) [](ancestors = 277909842)

Reply:
The `/login` literally means to use the exact string `/login` here (or directly assigned to [line 50](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/40/files#diff-3a67f168fce508486b1b71721472b542R50)). Those host and port can be omitted because the [HTTP Location header supports using relative url](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location#Directives), which we want here, so that we do not need to hardcode a configurable port number.

On the other hand, the path `/login` will not be in config file because it is not something can be or need to be configured. It needs to match [this hard-coded line](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/40/files#diff-3a67f168fce508486b1b71721472b542R54), so it is a self-contained business logic of this sample app.